agent/consul/leader.go

```diff
@@ -349,32 +348,48 @@ func (s *Server) reconcileReaped(known map[string]struct{}) error {
 			continue
 		}
 
-		// Create a fake member
-		member := serf.Member{
-			Name: check.Node,
-			Tags: map[string]string{
-				"dc":   s.config.Datacenter,
-				"role": "node",
-			},
-		}
-
 		// Get the node services, look for ConsulServiceID
 		_, services, err := state.NodeServices(nil, check.Node)
 		if err != nil {
 			return err
 		}
 		serverPort := 0
+		serverAddr := ""
+		serverID := ""
+
 		for _, service := range services.Services {
 			if service.ID == structs.ConsulServiceID {
 				serverPort = service.Port
+				_, node, err := state.GetNode(check.Node)
+				if err != nil {
+					s.logger.Printf("[ERR] Unable to look up node with name %v, got error:%v", check.Node, err)
```

Comment:
If we can't get an address we should probably just bail rather than let the `net.ParseIP` return something bogus below. I'd add a label and do a `continue CHECKS` here for the error.